runtime/lua/vim/img/_backend/kitty.lua

```diff
@@ -0,0 +1,108 @@
+---@class vim.img.KittyBackend: vim.img.Backend
+local M = {}
+
+---For kitty, we need to write an image in chunks
+---
+---Graphics codes are in this form:
+---
+---    <ESC>_G<control data>;<payload><ESC>\
+---
+---To stream data for a PNG, we specify the format `f=100`.
+---
+---To simultaneously transmit and display an image, we use `a=T`.
+---
+---Chunking data (such as from over a network) requires the
+---specification of `m=0|1`, where all chunks must have a
+---value of `1` except the very last chunk.
+---@param data string
+local function write_seq(data)
+  local terminal = require('vim.img._terminal')
+
+  terminal.write(terminal.code.ESC .. '_G') -- Begin sequence
```

Comment:
I don't think we need a special constant for `ESC`. We can just write `\027_G` here. That's what we already do everywhere else and it's fine.

If we were going to make anything into a constant, it should be the entire APC prefix (`ESC _`) and likewise for ST (`ESC \`), CSI, OSC, etc.